packages/react-ui-ag/src/Listings/Listing.js

```diff
@@ -0,0 +1,287 @@
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import autobind from 'autobind-decorator'
+import get from 'lodash/get'
+import LazyLoad, { forceCheck } from 'react-lazyload'
+import { Button, ToggleButton, ListingComponents, ListingCell } from '@rentpath/react-ui-core'
+import { Banner } from '../Banners'
+
+const buttonPropTypes = PropTypes.shape({
+  children: PropTypes.node,
+  onClick: PropTypes.func,
+  className: PropTypes.string,
+  valueLocation: PropTypes.string,
+})
+
+const REACT_LAZYLOAD_PROP_DEFAULTS = {
+  offset: [250, -100],
+  resize: true,
+  width: 318,
+  height: 280,
+}
+
+@themed(/^Listing/)
+export default class Listing extends Component {
+  static propTypes = {
+    index: PropTypes.number,
+    listing: PropTypes.object,
+    theme: PropTypes.object,
+    className: PropTypes.string,
+    onClick: PropTypes.func,
+    photos: PropTypes.object,
+    ratings: PropTypes.object,
+    ctaButtons: PropTypes.arrayOf(buttonPropTypes),
+    favoriteButton: buttonPropTypes,
+    lazyLoad: PropTypes.oneOfType([
+      PropTypes.object,
+      PropTypes.bool,
+    ]),
+    isActive: PropTypes.bool,
+    listingInfoComponents: PropTypes.func,
+  }
+
+  static defaultProps = {
+    isActive: true,
+    lazyLoad: true,
+    theme: {},
+    listing: {},
+    ratings: {},
+    photos: {},
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      loadedCarousel: false,
+    }
+  }
+
+  shouldComponentUpdate(nextProps) {
+    return this.props.isActive !== nextProps.isActive ||
+    this.props.listing.id !== nextProps.listing.id
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.isActive !== prevProps.isActive && !prevProps.isActive) {
+      forceCheck()
+    }
+  }
+
+  get listingPhotos() {
+    return get(this.props.listing, 'photos', [])
+  }
+
+  @autobind
+  handleCardClick() {
+    const { index, listing, onClick } = this.props
+
+    if (onClick) onClick(index, listing)
+  }
+
+  @autobind
+  handleButtonClick(onClick) {
+    const { isActive, listing } = this.props
+
+    return event => {
+      if (isActive && onClick) {
+        onClick(this.props.listing)
+      } else if (!isActive && this.props.onClick) {
+        this.props.onClick(this.props.index, listing)
+      }
+
+      const shouldStopPropagation =
+        isActive && event && event.stopPropagation
+
+      if (shouldStopPropagation) event.stopPropagation()
+    }
+  }
+
+  @autobind
+  handleFavoriteClick(value, event) {
+    const { isActive, listing, favoriteButton } = this.props
+    const { onClick } = favoriteButton
+
+    if (isActive && onClick) {
+      onClick(this.props.listing, value)
+    } else if (!isActive && this.props.onClick) {
+      this.props.onClick(this.props.index, listing)
+    }
+
+    const shouldStopPropagation = isActive && event && event.stopPropagation
+
+    if (shouldStopPropagation) {
+      event.stopPropagation()
```

Comment:
Will this break tagging?

Reply:
Good question, I'm not sure. I copied this function from `MobileMapListing` and this component is currently being used on mobile.

Reply:
@tadjohnston what do you think?

Reply:
It all depends on how the tag is fired.  If it's the normal way, no, because the `stopPropagation` here is related to the synthetic React event while the tagging event is completely outside of React.